cachito/web/api_v1.py

```diff
@@ -211,15 +213,25 @@ def download_archive(request_id):
         )
         raise InternalServerError()
 
+    hasher = hash_file(bundle_dir.bundle_archive_file)
+    checksum = hasher.hexdigest()
+    store_checksum = bundle_dir.bundle_archive_checksum.read_text(encoding="utf-8")
+    if checksum != store_checksum:
+        msg = "Checksum of bundle archive {} has changed."
+        flask.current_app.logger.error(msg.format(bundle_dir.bundle_archive_file))
+        raise InternalServerError(msg.format(bundle_dir.bundle_archive_file.name))
+
     flask.current_app.logger.debug(
         "Sending the bundle at %s for request %d", bundle_dir.bundle_archive_file, request_id
     )
-    return flask.send_file(
+    resp = flask.send_file(
         str(bundle_dir.bundle_archive_file),
         mimetype="application/gzip",
         as_attachment=True,
         attachment_filename=f"cachito-{request_id}.tar.gz",
     )
+    resp.headers["Digest"] = f"sha-256={base64.b64encode(hasher.digest())}"
```

Comment:
Minor problem, when I look at the headers locally with `curl -v localhost:8080/api/v1/requests/1/download`, I see this:
```
Digest: sha-256=b'UKVLAQIq0YRm4gY3VSV1avgwTZoAUvmeTbqDuAxnr24='
```
I think we want to `.decode()` the bytes, not just convert to str.

Optional - do you think it would make sense to have a function in the `checksum` module for converting digests to the base64 encoded value?

Reply:
> Optional - do you think it would make sense to have a function in the checksum module for converting digests to the base64 encoded value?

Absolutely.

Reply:
@chmeliik Both are fixed. PTAL. Thank you very much for catching this bug.